tsdapiclient/tacl2.py

```diff
@@ -63,6 +63,33 @@ def get_guide_options(ctx, args, incomplete):
     return [k for k,v in GUIDES.items() if incomplete in k]
 
 
+def get_dir_contents(ctx, args, incomplete):
+    sep = os.path.sep
+    if os.path.lexists(incomplete):
+        if os.path.isdir(incomplete):
+            if not incomplete.endswith(sep):
+                return [f'{incomplete}{sep}'] # prepare to list
+            else:
+                base, fragment = os.path.dirname(incomplete), os.path.basename(incomplete)
+                return [f'{base}{sep}{entry}' for entry in os.listdir(base) if fragment in entry]
+        elif os.path.isfile(incomplete):
+            base, fragment = os.path.dirname(incomplete), os.path.basename(incomplete)
+            return [incomplete]
+        else:
+            return [] # not sure what this could be (yet)
+    else:
+        if incomplete == '':
+            return os.listdir('.')
+        else:
+            base, fragment = os.path.dirname(incomplete), os.path.basename(incomplete)
```

Comment:
If we use `os.path.dirname(os.path.expanduser(incomplete))` here we can support referencing the home folder with tilde, and the path will be rewritten to start with the user's home. Alternately one could instead expand when doing `os.listdir(base)` below.

Trying to expand "~/" will however cause a NoneError, I'm thinking it might have to be a separate case? (`elif incomplete == '~/'`)

Reply:
Strategy is: `~` -> `$HOME`, and `~/` -> `$HOME/`. Easiest to reason about in code imho...